src/FormSerializationWriter.cs

```diff
@@ -162,18 +173,27 @@ public void WriteCollectionOfPrimitiveValues<T>(string? key, IEnumerable<T>? val
     {
         if(depth > 0) throw new InvalidOperationException("Form serialization does not support nested objects.");
         depth++;
-        if(value == null && !additionalValuesToMerge.Any(static x => x != null)) return;
+
+        if (value == null)
+        {
+            using (var enumerator = additionalValuesToMerge.GetEnumerator())
+            {
+                if (!enumerator.MoveNext()) return;
+            }
+        }
```

Comment:
I think this is still different from the original. There should be a foreach loop looking for x != null.